src/libcore/result.rs

```diff
@@ -309,6 +309,30 @@ impl<T, E> Result<T, E> {
         !self.is_ok()
     }
 
+    /// Returns `true` if the result is an [`Ok`] value containing the given value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let x: Result<u32, &str> = Ok(2);
+    /// assert_eq!(x.contains(&2), true);
+    ///
+    /// let x: Result<u32, &str> = Ok(3);
+    /// assert_eq!(x.contains(&2), false);
+    ///
+    /// let x: Result<u32, &str> = Err("Some error message");
+    /// assert_eq!(x.contains(&2), false);
+    /// ```
+    #[must_use]
+    #[inline]
+    #[stable(feature = "result_contains", since = "???")]
+    pub fn contains(&self, x: &T) -> bool where T: PartialEq {
+        match self {
+            Ok(y) => y == x,
+            Err(_) => false
+        }
+    }
```

Comment:
Perhaps add `pub fn contains_err(&self, x: &E) -> bool where E: PartialEq {` for symmetry?

Reply:
Done! Thanks!